crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -423,6 +428,14 @@ where
         to_block: u64,
     ) -> Result<Vec<Log>, FilterError> {
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
+        let is_multi_block_range = from_block != to_block;
+
+        if is_multi_block_range
+            && self.max_blocks_per_filter != 0
+            && to_block - from_block > self.max_blocks_per_filter
```

Comment:
this can underflow since unvalidated user input

Reply:
@mattsse I supposed that to_block will be always bigger than from_block since I assumed it is already filtered by `logs_utils::get_filter_block_range`. Am I missing something here or should I also check if from_block is lower than to_block?

Reply:
ah I think you're right